src/sage/schemes/elliptic_curves/hom_composite.py

```diff
@@ -893,3 +894,290 @@ def inseparable_degree(self):
+
+    - ``ell`` -- an integer defining the torsion `E[\ell]`.
+
+    - ``use_second_curve`` (optional, default: ``True``): whether to base
+      the remaining torsion representation to the second curve (which is faster
+      because of lower degree).
+
+
+    OUTPUT: a polynomial dividing the division polynomial of the
+    starting curve (or of the second curve if `use_second_curve` is set),
+    representing the torsion subgroup that is not nullified by the isogeny.
+
+    TESTS::
+
+        sage: from sage.schemes.elliptic_curves.hom_composite import _sieve_torsion_through_isogenies
+
+        sage: _sieve_torsion_through_isogenies([], 2)
+        Traceback (most recent call last):
+        ...
+        ValueError: list of isogenies must be nonempty
+
+        sage: E0 = EllipticCurve(GF(37), [4, 0])
+        sage: phi = E0.scalar_multiplication(0)
+        sage: _sieve_torsion_through_isogenies([phi], 2)
+        1
+        sage: phi = E0.scalar_multiplication(2)
+        sage: _sieve_torsion_through_isogenies([phi], 2)
+        1
+        sage: phi = E0.isogeny(E0(0,0)); phi.kernel_polynomial()
+        x
+        sage: _sieve_torsion_through_isogenies([phi], 2) == phi.dual().kernel_polynomial()
+        True
+        sage: _sieve_torsion_through_isogenies([phi], 2, use_second_curve=False).monic()
+        x^2 + 4
+        sage: div = E0.division_polynomial(2); (div // phi.kernel_polynomial()).monic()
+        x^2 + 4
+        sage: _sieve_torsion_through_isogenies([phi], 3).degree()
+        4
+
+    .. SEEALSO::
+
+        - :meth:`sage.schemes.elliptic_curves.hom_composit.EllipticCurveHom_composite.is_cyclic`
+    """
+    if not phis:
+        raise ValueError("list of isogenies must be nonempty")
+    if any(phi.is_zero() or not phi.is_separable() for phi in phis):
+        return phis[0].domain().base_ring()['x'].one()
+
+    E0 = phis[0].domain()
+    p = E0.base_ring().characteristic()
+
+    # evalute isogenies step by step
+    # modulo the division polynomial of the starting curve
+    # i.e. "working with full torsion" at the same time
+    # gradually removing torsion factors that are being nullified
+    mod = E0.division_polynomial(ell)
+    if p.gcd(ell) != 1:
+        # if the order is not coprime with the characteristic,
+        # we may get repeated roots
+        mod = mod.radical()
+
+    # x : the x-coordinate of the generic ell-torsion point
+    # on the starting curve
+    # y : current value of isogenies applied to x
+    x = mod.parent().quotient(mod).gen()
+    y = x
+
+    for i, phi in enumerate(phis):
+        x_rational_map = phi.x_rational_map()
+
+        den = x_rational_map.denominator()(y)
+        g = den.lift().gcd(mod)
+        if g != 1:
+            # a new part of torsion is nullified
+            # so we need to update the modulus and the quotient ring
+            mod //= g
+            assert mod
+            if mod.degree() <= 0:  # just 1? or -1?
```

Comment:
```suggestion
            if mod.is_constant():
```
(Not sure what the comment means.)